src/controls/controller.h

```diff
@@ -16,11 +36,12 @@ enum ControllerDirection {
 
 void controllersClearState();
 int controllerIsConnected(int index);
-OSContPad* controllersGetControllerData(int index);
-u16 controllerGetLastButton(int index);
-u16 controllerGetButton(int index, u16 button);
-u16 controllerGetButtonDown(int index, u16 button);
-u16 controllerGetButtonUp(int index, u16 button);
+int8_t   controllerGetStickX(int index);
```

Comment:
I like your idea of not exposing the entire state and instead requiring use of the dedicated functions.

In this case, I think it's worth the convenience of returning both joystick axis positions at once - more often than not, that's what callers will need. Can you create a struct containing both axis positions and use that instead?

Reply:
Done in 35fcffe.

> I like your idea of not exposing the entire state and instead requiring use of the dedicated functions.

Initially the goal was to remove the dependency on libultra's `OSContPad`. Then I realised that the only values used were the x and y axes of the stick, and I thought about having a single getter for both, but in the end the fact that in one case the game only needs to access one of the axes made me stick (no pun intended) to the two-getter implementation.